### PR TITLE
feature: automatically ugrade on latest snapshot

### PR DESCRIPTION
**Pull request by dos65**
This add an option to automatically upgrade server version on the latest
SNAPSHOT once a day.

New setting name: `autoLatestSnapshot`

Requires: https://github.com/scalameta/metals/pull/3568

### CODE REVIEW

File: src/checkServerVersion.ts [hunk 1]
```diff
@@ -82,3 +123,120 @@ function serverVersionInfo(config: WorkspaceConfiguration): {
...
+        if (text.endsWith("/")) {
+          const parsed = parseMetalsVersion(text);
+          return parsed ? [parsed] : [];
+        } else {
+          return [];
+        }
+      } else {
+        return [];
+      }
+    });
```
Comment (kpodsiad) [inline src/checkServerVersion.ts:154]: I wonder if we could add some marker class to `td` element and then extract it string using regex `<td class="marker">.*<\/td>`

**Review by kpodsiad — COMMENTED** [review]
I took only a quick look and left few comments

**Comment by gabro** [comment]
Just passing by, but have you considered publishing a "true" mutable snapshot called `SNAPSHOT` (or `0.12-SNAPSHOT`, or similar) and then users can opt-in by setting the Scala version to `SNAPSHOT`?

This is similar to how local development works and it has the added benefit of relying on the existing infrastructure.

**Comment by tgodzik** [comment]
> Just passing by, but have you considered publishing a "true" mutable snapshot called `SNAPSHOT` (or `0.12-SNAPSHOT`, or similar) and then users can opt-in by setting the Scala version to `SNAPSHOT`? This is of course in addition to regular immutable snapshots.
> 
> This is similar to how local development works and it has the added benefit of relying on the existing infrastructure.

I think that would probably be awesome to have, no need to tinker with dates etc. What do you think @dos65  ?

**Comment by dos65** [comment]
@gabro @tgodzik I lost in Pr's.
I haven't thought about mutable SNAPSHOT until @gabro left comment but I don't think that it's the best option.
My option about it here - https://github.com/scalameta/metals/pull/3568#issuecomment-1022089111

**Pull request closed without merging**